llm/memory.go

```diff
@@ -66,7 +66,7 @@ type MemoryEstimate struct {
 
 // Given a model and one or more GPU targets, predict how many layers and bytes we can load, and the total size
 // The GPUs provided must all be the same Library
-func EstimateGPULayers(gpus []gpu.GpuInfo, ggml *GGML, projectors []string, opts api.Options) MemoryEstimate {
+func EstimateGPULayers(gpus []gpu.GpuInfo, ggml *GGML, projectors []string, opts *api.Options) MemoryEstimate {
```

Comment:
I noticed here and a few other places below `opts` has become a pointer. To avoid accidentally mutating it can we keep it pass by value?

Reply:
fixed